src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointClusterer.java

```diff
@@ -1,61 +1,70 @@
 package org.broadinstitute.hellbender.tools.spark.sv;
 
+import org.apache.spark.api.java.function.FlatMapFunction;
+import org.apache.spark.api.java.function.Function;
+
 import java.util.*;
-import java.util.function.Function;
 
 /**
  * A class that acts as a filter for breakpoint evidence.
  * It passes only that evidence that is part of a putative cluster.
  */
-public final class BreakpointClusterer implements Function<BreakpointEvidence, Iterator<BreakpointEvidence>> {
+public final class BreakpointClusterer implements Iterator<BreakpointEvidence> {
     private final int minEvidenceCount;
     private final int staleEventDistance;
-    private final SortedMap<BreakpointEvidence, Boolean> locMap = new TreeMap<>();
-    private final List<Map.Entry<BreakpointEvidence, Boolean>> reportableEntries;
-    private static final Iterator<BreakpointEvidence> noEvidence = Collections.emptyIterator();
-    private int currentContig = -1;
+    private final SVIntervalTree<List<BreakpointEvidence>> evidenceTree;
+    private Iterator<SVIntervalTree.Entry<List<BreakpointEvidence>>> treeItr;
+    private Iterator<BreakpointEvidence> listItr;
 
     public BreakpointClusterer( final int minEvidenceCount, final int staleEventDistance ) {
         this.minEvidenceCount = minEvidenceCount;
         this.staleEventDistance = staleEventDistance;
-        this.reportableEntries = new ArrayList<>(2 * minEvidenceCount);
+        this.evidenceTree = new SVIntervalTree<>();
     }
 
-    @Override
-    public Iterator<BreakpointEvidence> apply( final BreakpointEvidence evidence ) {
-        if ( evidence.getContigIndex() != currentContig ) {
-            currentContig = evidence.getContigIndex();
-            locMap.clear();
+    public Iterator<BreakpointEvidence> apply( final Iterator<BreakpointEvidence> evidenceItr ) {
+        while ( evidenceItr.hasNext() ) {
+            final BreakpointEvidence evidence = evidenceItr.next();
+            final SVInterval location = evidence.getLocation();
+            final SVIntervalTree.Entry<List<BreakpointEvidence>> entry = evidenceTree.find(location);
+            if ( entry != null ) entry.getValue().add(evidence);
```

Comment:
Pretty sure that Louis said that this was one of our departures from Google style:  single statements following an "if", "else", or "else if" that fit comfortably on the same line are allowed (but not required) to be unbraced.
Since you prefer braces, I'll change these.
However, since you've thrown down the gauntlet, I'm going to start nailing you guys on very long lines (max line length is supposed to be 100 characters).  So there.